From 26a85d7e00fd66b18c7fd90dd25ab8255939902c Mon Sep 17 00:00:00 2001
From: Andrii Osipov <aosipov@anaconda.com>
Date: Thu, 26 Mar 2026 10:29:54 -0700
Subject: [PATCH] Prevent mksnapshot run with unlimited threads

---
 src/3rdparty/chromium/v8/BUILD.gn | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/3rdparty/chromium/v8/BUILD.gn b/src/3rdparty/chromium/v8/BUILD.gn
index 4c6f7fb4b1..22264eb079 100644
--- a/src/3rdparty/chromium/v8/BUILD.gn
+++ b/src/3rdparty/chromium/v8/BUILD.gn
@@ -118,7 +118,8 @@ declare_args() {
   v8_enable_fast_mksnapshot = false
 
   # Enable using multiple threads to build builtins in mksnapshot.
-  v8_enable_concurrent_mksnapshot = true
+  # To prevent exhaust available RAM, leading to the -5 crash when mksnapshot is running.
+  v8_enable_concurrent_mksnapshot = false
 
   # Optimize code for Torque executable, even during a debug build.
   v8_enable_fast_torque = ""
-- 
2.39.3 (Apple Git-146)

