From 53690e2c164793bd1062fae746fe73129e2122d6 Mon Sep 17 00:00:00 2001
From: Brian Wing <bwing@anaconda.com>
Date: Wed, 4 Dec 2024 14:49:21 -0500
Subject: [PATCH] Add gn args for libxslt and libwebp

---
 src/core/CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index d5e9b71..2b30432 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -515,6 +515,14 @@ foreach(arch ${archs})
                 CONDITION QT_FEATURE_webengine_vaapi
             )
             extend_gn_list(gnArgArg
+                    ARGS use_system_libxslt
+                    CONDITION QT_FEATURE_webengine_system_libxslt
+            )
+            extend_gn_list(gnArgArg
+                    ARGS use_system_libwebp
+                    CONDITION QT_FEATURE_webengine_system_libwebp
+            )
+            extend_gn_list(gnArgArg
                 ARGS ozone_platform_x11 use_xkbcommon
                 CONDITION QT_FEATURE_webengine_ozone_x11
             )
-- 
2.45.2
